src/core/net/ip6.cpp

```diff
@@ -414,7 +414,7 @@ ThreadError Ip6::HandleDatagram(Message &message, Netif *netif, uint8_t interfac
 
 exit:
 
-    if (error == kThreadError_Drop)
+    if ((error == kThreadError_Drop) || (error == kThreadError_NoRoute))
```

Comment:
1. ForwardMessage may set error to kThreadError_NoRoute. Is my understanding right?
2. I will submit patch to make it a general catch-all error.